CMakeLists.txt

```
@@ -26,6 +26,11 @@ CMAKE_POLICY(SET CMP0020 NEW)
 # Request C++17 Standard Support (Required for std::optional and auto types in lambda parameters)
 set(CMAKE_CXX_STANDARD 17)
 
+# Include Conan Packages
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
```

```suggestion
conan_basic_setup(CMAKE_TARGETS)
```

We should be using modern targets where possible in CMake.